server/sonar-server/src/main/java/org/sonar/server/computation/ws/SubmitReportWsAction.java

```diff
@@ -72,16 +68,21 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkGlobalPermission(GlobalPermissions.SCAN_EXECUTION);
     String projectKey = request.mandatoryParam(PARAM_PROJECT_KEY);
-    long snapshotId = request.mandatoryParamAsLong(PARAM_SNAPSHOT);
-    try (InputStream reportData = request.paramAsInputStream(PARAM_REPORT_DATA)) {
-      String reportKey = queue.add(projectKey, snapshotId, reportData);
+    InputStream reportData = request.paramAsInputStream(PARAM_REPORT_DATA);
+    try {
+      ReportQueue.Item item = queue.add(projectKey, reportData);
       workerLauncher.startAnalysisTaskNow();
       response.newJsonWriter()
         .beginObject()
-        .prop("key", reportKey)
+        // do not write integer for forward-compatibility, for example
+        // if we want to write UUID later
+        .prop("key", String.valueOf(item.dto.getId()))
```

Comment:
web console still displays ids, so I didn't want to change behavior in last minutes of sprint. Let's do that in 5.2.